ledger/src/shred/merkle.rs

```diff
@@ -434,9 +589,133 @@ fn make_merkle_branch(
     Some(MerkleBranch { root, proof })
 }
 
+pub(super) fn recover(mut shreds: Vec<Shred>) -> Result<Vec<Shred>, Error> {
+    // Grab {common, coding} headers from first coding shred.
+    // TODO: Verify that other shreds are consistent!
```

Comment:
are there additional consistency checks to add?

Reply:
added consistency checks for shreds headers.